### DISCUSSION TITLE
@ngrx/store or @ngrx/signals for angular signals

### DISCUSSION THREAD

**Discussion by ls-infra** [ngrx/platform#4555] [Q&A]
hello , if i use https://angular.dev/guide/signals for the new angular project , 
should i use it with `@ngrx/store` or  `@ngrx/signals`  ?

thank you very much ,

**Comment by rainerhahnekamp** [comment]
@ls-infra if it comes down to a new project, I would first go with `@ngrx/signals`? Why? Because it is more lightweight and you might not need the Redux pattern from `@ngrx/store`.

Once you have the need for eventing, that is your components should be able to dispatch events that could be consumed by multiple stores, then you have the need for `@ngrx/store`.

**Reply by jits** [reply]
Hi @rainerhahnekamp — would you say that the [`withRedux` extension for SignalStore (from your NgRx Toolkit)](https://github.com/angular-architects/ngrx-toolkit?tab=readme-ov-file#redux-withredux) is a good stepping stone from plain SignalStore to a more event-based architecture?

**Reply by rainerhahnekamp** [reply]
@jits `withRedux` comes without global event dispatching. So we don't have the level of decoupling compared to the global store. There were some plans for a rewrite, but since @markostanimirovic will present an events extension that contains the global dispatching, we are waiting for what will happen. If Marko's extensions gets merged into the SignalStore, I don't see any reason to start a new project with the global store. If it stays a community extension, we might have the option to merge it into the toolkit. At least, that's my hope.

**Reply by jits** [reply]
Thanks @rainerhahnekamp for the clarification! Looking forward to seeing what @markostanimirovic and you (and the rest of the team) come up with for the events extension (assuming it's intended to be part of core).

**Reply by markostanimirovic** [reply]
SignalStore `events` plugin prototype is in the following repo: https://github.com/markostanimirovic/ngrx-signals-events-prototype

**Reply by jits** [reply]
@markostanimirovic — thanks! What are the next steps with that project/initiative?
